platform/lib/pipeline/componentReferenceImporter.js

```diff
@@ -224,11 +223,21 @@ class ComponentReferenceImporter {
     const tag = this._findExtensionTag(extension.name) || {};
     const script = this._findExtensionScript(extension.name) || {};
 
-    spec.version = spec.version.filter((version) => version != LATEST_VERSION);
+    // Parse available versions from file system
+    const versions = new Set();
+    for (const file of extension.files) {
+      const path = file.substring(`extensions/${spec.name}/`.length);
+      const match = path.match(/^(\d+\.\d+)\//);
+      if (match) {
+        versions.add(match[1]);
+      }
+    }
+    spec.version = Array.from(versions);
+
+    // Calculate the latest version
```

Comment:
There is a concept of a "latest version" of a component that is specified here in `bundles.config.js`. amp-carousel is an example of when the `latestVersion` specified is not always the larger: https://github.com/ampproject/amphtml/blob/91aaf6f3e28c28117fc6ca0eb660d39161e387ee/build-system/compile/bundles.config.js#L460-L464

Is this relevant at all here?

Reply:
Thanks! Afaik, we use the experimental flag instead. But we should use this information instead of auto calculating the latest version.

//cc @matthiasrohmer

Reply:
Created #5200 to track this, thanks for the hint Caroline!